### PR TITLE
fix(iceberg): fix iceberg engine table time travel

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- https://github.com/risingwavelabs/risingwave/issues/19418
- Fix time travel for iceberg. Iceberg supports different time travel syntax from hummock table, so we should check the syntax accordingly.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by stdrc** [comment]
> * Iceberg supports different time travel syntax from hummock table.

So is it changed to have the same syntax or not?

**Comment by chenzl25** [comment]
> > * Iceberg supports different time travel syntax from hummock table.
> 
> So is it changed to have the same syntax or not?

They don't share the same syntax. For example, hummock table time travel doesn't support `FOR SYSTEM_VERSION AS OF 123456` while iceberg table doesn't support `FOR SYSTEM_TIME AS OF PROCTIME()`

File: e2e_test/iceberg/test_case/iceberg_engine.slt [hunk 1]
```diff
@@ -15,6 +15,11 @@ select * from t;
 ----
 1 xxx
 
+query ??
+select * from t for system_time as of '2222-12-10 11:48:06';
```
Comment (xiangjinwu) [inline e2e_test/iceberg/test_case/iceberg_engine.slt:19]: Does `'2222-12-10 11:48:06'` correspond to `7982020086000` or `7981991286000`?

Comment (chenzl25) [reply e2e_test/iceberg/test_case/iceberg_engine.slt:19]: I look at the implementation, currently all timestamp string without explicit timezone will be considered to be UTC and don't respect the session timezone.

**Review by zwang28 — APPROVED** [review]

File: e2e_test/iceberg/test_case/iceberg_engine.slt (hunk 1 above)
Comment (chenzl25) [reply e2e_test/iceberg/test_case/iceberg_engine.slt:19]: Let's fix the current issue first and I will open another issue to track the timezone issue.

Comment (chenzl25) [reply e2e_test/iceberg/test_case/iceberg_engine.slt:19]: https://github.com/risingwavelabs/risingwave/issues/19757

**Pull request merged**